patches/content-browser-permissions-permission_service_impl.h.patch

```diff
@@ -0,0 +1,14 @@
+diff --git a/content/browser/permissions/permission_service_impl.h b/content/browser/permissions/permission_service_impl.h
+index 85359a821fb4641a1bc904e1f3c0adcb54a3ed40..27f0fd0ed9f6edb5703232d19e6bb2752e75c6b6 100644
+--- a/content/browser/permissions/permission_service_impl.h
++++ b/content/browser/permissions/permission_service_impl.h
+@@ -44,6 +44,9 @@ class CONTENT_EXPORT PermissionServiceImpl
+   // blink::mojom::PermissionService.
+   void HasPermission(blink::mojom::PermissionDescriptorPtr permission,
+                      PermissionStatusCallback callback) override;
++  bool HasPermission(blink::mojom::PermissionDescriptorPtr permission,
```

Comment:
do we really need to do this?
In generated `permission.mojom.h`
```
170   // Sync method. This signature is used by the client side; the service side
171   // should implement the signature with callback below.
172   virtual bool HasPermission(PermissionDescriptorPtr permission, ::blink::mojom::PermissionStatus* out_status);
173
174   using HasPermissionCallback = base::OnceCallback<void(::blink::mojom::PermissionStatus)>;
175   virtual void HasPermission(PermissionDescriptorPtr permission, HasPermissionCallback callback) = 0;
```

Reply:
Yes, because when the renderer needs to know (in `BraveContentSettingsObserver::AllowAutoplay`) whether to allow autoplay, we need to wait for an answer from the autoplay whitelist service. We have no mechanism for going through the DOM and changing the answer later.

Reply:
The above generated header is what I added "Sync" property to the mojo file.
So what I'm saying is, you can add Sync without manual adding implementation.
You can see other mojo file with Sync property in our Brave repo like rewards.

Reply:
fixed, thank you